dove/src/transaction.rs

```diff
@@ -0,0 +1,875 @@
+
+        if meta.type_parameters.len() != self.type_parameters.len() {
+            return Err(anyhow!(
+                "Script '{}' takes {} type parameters, {} passed",
+                meta.name,
+                meta.type_parameters.len(),
+                self.type_parameters.len()
+            ));
+        }
+
+        let (signers, args_count, args) = self.prepare_arguments(&meta.parameters)?;
+
+        if self.args.len() != args_count {
+            return Err(anyhow!(
+                "Script '{}' takes {} parameters, {} passed",
+                meta.name,
+                args_count,
+                self.args.len()
+            ));
+        }
+
+        Ok((
+            meta.name,
+            Transaction::new(signers as u8, unit, args, self.type_parameters.clone())?,
+        ))
+    }
+    /// Find and run the script using the specified parameters
+    pub fn run(&self) -> Result<PipelineExecutionResult, Error> {
+        let (script, _meta) = self.lookup_script()?;
+        let dep_list = self.get_dep_list(script.name().as_ref())?;
+        let executor = Executor::new(self.dove_ctx.dialect.as_ref(), self.signers[0], dep_list);
+        executor.execute_script(script, Some(self.signers.clone()), self.args.clone())
+    }
+    // =============================================================================================
+    fn lookup_script_by_file_name(&self, fname: &str) -> Result<(MoveFile, Meta), Error> {
+        let script_path = self
+            .dove_ctx
+            .path_for(&self.dove_ctx.manifest.layout.scripts_dir);
+        let file_path = if !fname.ends_with("move") {
+            let mut path = script_path.join(fname);
+            path.set_extension("move");
+            path
+        } else {
+            script_path.join(fname)
+        };
+        if !file_path.exists() {
+            return Err(anyhow!("File [{}] not found", fname));
+        }
+
+        let sender = self.dove_ctx.account_address()?;
+        let script = MoveFile::load(&file_path)?;
+        let mut scripts =
+            ScriptMetadata::extract(self.dove_ctx.dialect.as_ref(), Some(sender), &[&script])?;
+        if scripts.is_empty() {
+            return Err(anyhow!("Script not found in file '{}'", fname));
+        }
+
+        let meta = if scripts.len() > 1 {
+            let mut scripts = scripts
+                .into_iter()
+                .filter(|sc| {
+                    if let Some(script_name) = &self.script_name {
+                        &sc.name == script_name
+                    } else {
+                        false
+                    }
+                })
+                .collect::<Vec<_>>();
+            if scripts.len() > 1 {
+                return Err(anyhow!(
+                    "There are several scripts with the name '{:?}' in file '{}'",
+                    self.script_name,
+                    fname
+                ));
+            } else {
+                scripts.remove(0)
```

Comment:
Why `remove` is used here?

Reply:
Comfortable. And in the source code it was also

Reply:
Please replace all `array.remove(idx)` with `array[0]`. I can't see a reason for using mutable (and not cheap at all!) operation instead of a simple element access.